receiver/prometheusreceiver/internal/otlp_transaction.go

```diff
@@ -157,6 +167,22 @@ func (t *transaction) Rollback() error {
 	return nil
 }
 
+func (t *transaction) AddTargetInfo(labels labels.Labels) error {
+	attrs := t.nodeResource.Attributes()
+
+	for _, lbl := range labels {
+		if lbl.Name == "job" || lbl.Name == "instance" || lbl.Name == model.MetricNameLabel {
+			continue
+		}
+
+		// TODO: Do I need to sanitise (desantise)?
```

Comment:
If sanitize/desanitize if necessary should do that now instead of later

Reply:
This is more because I couldn't figure out what to do here.

So when converting from attributes to labels, we replace any non-digit/non-alphabetic character to `_`. See: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/pkg/translator/prometheusremotewrite/helper.go#L187

This means an attribute like `http.scheme` would be converted to `http_scheme`. Now going the other way, should I convert `_` to `.` or not? I am not sure @dashpole

Reply:
Can this TODO be removed then?

Reply:
Done!